sympy/logic/tests/test_lra_theory.py

```diff
@@ -0,0 +1,306 @@
+    phi = (x >= 0) & (x >= 1) & (x <= -1)
+    cnf = CNF.from_prop(phi)
+    enc = EncodedCNF()
+    enc.from_cnf(cnf)
+    lra, x_subs, s_subs = LRASolver.from_encoded_cnf(enc)
+
+
+    assert len(lra.A) == 0
+    assert lra.assert_enc_boundry(enc.symbols.index((Q.ge(x, 0))) +1) == ("OK", None)
+    assert lra.assert_enc_boundry(enc.symbols.index((Q.ge(x, 1))) +1) == ("OK", None)
+    assert lra.assert_enc_boundry(enc.symbols.index((Q.le(x, -1))) + 1) == ('UNSAT', {-(enc.symbols.index((Q.le(x, -1))) + 1), -(enc.symbols.index((Q.ge(x, 1))) +1)})
+    # assert lra.assert_con(Q.ge(x, 0)) == ("OK", None)
+    # assert lra.assert_con(Q.ge(x, -1)) == ('SAT', {x: 0, y: 0})
+    # assert lra.assert_con(Q.le(x, -1)) == ('UNSAT', {x <= -1, x >= 0})
+    #
+    # m = Matrix()
+    # lra = LRASolver(m, [], [x, y])
+    # assert lra.assert_con(Q.le(x, -1)) == ("OK", None)
+    # assert lra.assert_con(Q.ge(x, 0)) == ('UNSAT', {x <= -1, x >= 0})
+    #
+    # m = Matrix([[-1, -1, 1, 0], [-2, 1, 0, 1]])
+    # #assert LRASolver._pivot(m, 0, 0) == Matrix([[1, 1, -1, 0], [0, 3, -2, 1]])
+    #
+    # # Example from page 89–90 of
+    # # "A Fast Linear-Arithmetic Solver for DPLL(T)"
+    # equations = [Eq(s1, -x + y), Eq(s2, x + y)]
+    # A, _ = linear_eq_to_matrix(equations, [x, y, s1, s2])
+    # A = -A # the identity matrix should be negative
+    # lra = LRASolver(A, [s1, s2], [x, y])
+    # assert lra.check() == ('SAT', {x: 0, y: 0, s1: 0, s2: 0})
+    # assert {v: lra.assign[v] for v in lra.all_var} == {x:0, y:0, s1:0, s2:0}
+    #
+    # assert lra.assert_con(x <= -4) == ("OK", None)
+    # assert lra.check() == ('SAT', {x: -4, y: 0, s1: 4, s2: -4})
+    # assert {v: lra.assign[v] for v in lra.all_var} == {x:-4, y:0, s1: 4, s2:-4}
+    #
+    # assert lra.assert_con(x >= -8) == ("OK", None)
+    # assert lra.check() == ('SAT', {x: -4, y: 0, s1: 4, s2: -4})
+    # assert {v: lra.assign[v] for v in lra.all_var} == {x:-4, y:0, s1:4, s2:-4}
+    #
+    # # note that this is the first time a pivot is used
+    # assert lra.assert_con(s1 <= 1) == ("OK", None)
+    # assert lra.check() == ('SAT', {x: -4, y: -3, s1: 1, s2: -7})
+    # assert {v: lra.assign[v] for v in lra.all_var} == {x:-4, y:-3, s1:1, s2:-7}
+    #
+    # assert lra.assert_con(s2 >= -3) == ("OK", None)
+    # assert lra.check() == ('UNSAT', {s1 <= 1, x <= -4, s2 >= -3})
+    #
+    #
+    # r1 = x <= 0
+    # r2 = y <= 0
+    # r3 = s1 >= 2
+    # equations = [Eq(s1, x+y)]
+    # A, _ = linear_eq_to_matrix(equations, [x, y, s1])
+    # A = -A  # the identity matrix should be negative
+    # lra = LRASolver(A, [s1], [x, y])
+    # lra.assert_con(x <= 0)
+    # lra.assert_con(y <= 0)
+    # lra.assert_con(s1 >= 2)
+    # assert lra.check() == ('UNSAT', {s1 >= 2, x <= 0, y <= 0})
+    #
+    #
+    # # test potential edge case
+    # r1 = x <= 1
+    # r2 = -x <= -5
+    # lra = LRASolver(Matrix(), [], [x])
+    # lra.assert_con(x <= 1)
```

Comment:
This code uses depreciated functions. I'm not sure if I should convert it into code that uses the new functions or just get rid of it. I feel like it's kind of pointless because test_random_problems() should be able to handle all of the cases here.